specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,544 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ComputeDiagnosticResourceProviderClient",
+    "description": "The Compute Diagnostic Resource Provider Client",
+    "version": "2021-06-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnosticOperations/{operationId}": {
+      "get": {
+        "tags": [
+          "diagnosticOperations"
+        ],
+        "operationId": "DiagnosticOperations_Read",
+        "description": "Gets status of a Diagnostic operation issued for Disk Inspection.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "in": "path",
+            "name": "operationId",
+            "required": true,
+            "type": "string",
+            "description": "The identifier pointing to a request for Disk Inspection. This is used for polling result of the request."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the status of a scheduled disk inspection request.": {
+            "$ref": "./examples/GetDiagnosticOperation.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/run": {
```

Comment:
the url should be defined as either `diagnostics/default` or `diagnostics/{diagnosticCategory}`, same as other places

Reply:
Hi Zhenhua, the POST call requires the following URL "diskInspection/run"
![image](https://github.com/Azure/azure-rest-api-specs/assets/18197375/9fbab57c-5c4e-48f1-b14e-6bbdec0607f1)

Should I still change it?

Reply:
I think this is OK: there are a few "singleton" instances of diagnostics, so they can't be all called default. Different names are needed, but they are static instances, so they also don't need to be parameterized on the Url. Also, "run" is fine as an action name. The only suggestion I have would be that "start" might be a better name, since the results are obtained asynchronously.

Reply:
Resolving based on comment